src/index.ts

```diff
@@ -44,13 +44,18 @@ export type Value =
 type ValueSet = Set<Value>;
 type ValueMap = Map<Value, Value>;
 
+export interface Options {
+  jsonFallback?: boolean;
+}
+
 /**
  * Convert a value to an ESTree node
  *
  * @param value - The value to convert
+ * @param options
  * @returns The ESTree node.
  */
-export function valueToEstree(value?: Value): Expression {
+export function valueToEstree(value?: Value, options?: Options): Expression {
```

Comment:
I prefer to use a default value over checking if it’s defined later.

```suggestion
export function valueToEstree(value?: Value, options: Options = {}): Expression {
```

Since the type can be basically be anything now, I think it makes to sense to accept an `unknown` value if `options` is defined.

```suggestion
export function valueToEstree(value?: Value): Expression;
export function valueToEstree(value: unknown, options: Options): Expression;
export function valueToEstree(value?: Value, options: Options = {}): Expression {
```

Or perhaps simply:

```suggestion
export function valueToEstree(value?: unknown, options: Options = {}): Expression {
```

and then remove the`Value` type.

Reply:
Alternatively an overload? `unknown` if `OptionsWithJson` or so, and `Value` otherwise?
I’m up for both

Reply:
Let’s just use `unknown` without the overload. Basically anything is supported except functions anyway.